packages/@azure/servicebus/data-plane/lib/namespace.ts

```diff
@@ -205,13 +205,17 @@ export class Namespace {
       throw new Error("'tokenProvider' is a required parameter and must be of type: 'object'.");
     }
     if (!host.endsWith("/")) host += "/";
-    const connectionString =
-      `Endpoint=sb://${host};SharedAccessKeyName=defaultKeyName;` +
-      `SharedAccessKey=defaultKeyValue`;
-    if (!options) options = {};
-    const nsOptions: NamespaceOptions = options;
-    nsOptions.tokenProvider = tokenProvider;
-    return Namespace.createFromConnectionString(connectionString, nsOptions);
+    const config: ConnectionConfig = {
+      connectionString: "",
+      endpoint: `sb://${host}`,
+      host,
+      sharedAccessKeyName: "defaultKeyName",
+      sharedAccessKey: "defaultKeyValue"
+    };
+    config.connectionString = `Endpoint=${config.endpoint};SharedAccessKeyName=${
```

Comment:
nit: since it is still needed to assemble the connection string, maybe just use `ConnectionConfig.create()`.

```typescript
    const connectionString =
      `Endpoint=sb://${host};SharedAccessKeyName=defaultKeyName;` +
      `SharedAccessKey=defaultKeyValue`;
    const config = ConnectionConfig.create(connectionString);
```

Seems no tests are affected. No existing test coverage or not needed?

Reply:
> since it is still needed to assemble the connection string, maybe just use ConnectionConfig.create().

good point, updated

> Seems no tests are affected. No existing test coverage or not needed?

Yes